### ISSUE TITLE
deno compile ignores custom ca-certificates on linux when downloading runtimes

### ISSUE THREAD

**Issue by dshick-nydig** [denoland/deno#15921]
I am using the official `denoland/deno:alpine` docker image to build some cross-platform binaries.

The build must run behind a proxy that performs SSL re-encryption, and requires a custom cert to be trusted. I've added this cert to the docker image like so:

```Dockerfile
FROM denoland/deno:alpine-1.25.2

ADD .certs/dummy .certs/*.crt /usr/local/share/ca-certificates/
RUN chmod 644 /usr/local/share/ca-certificates/*.crt \
    && cat /usr/local/share/ca-certificates/*.crt \
    >> /etc/ssl/certs/ca-certificates.crt
    
...
```

I'm trying to run something like:

```sh
deno --unstable compile --cert="/etc/ssl/certs/ca-certificates.crt" -o build/mod-x86_64-apple-darwin \
        --target x86_64-apple-darwin \
        mod.ts
```

Without the `--cert` arg, the build fails much earlier when downloading imports. With the `--cert` arg, it successfully downloads and caches the imports, but fails to download the runtime binary.

```
Checking https://dl.deno.land/release/v1.25.2/deno-x86_64-apple-darwin.zip
Sending fatal alert BadCertificate
error: error sending request for url (https://dl.deno.land/release/v1.25.2/deno-x86_64-apple-darwin.zip): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer

Caused by:
    0: error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
    1: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
```

I've also tried using the following in my dockerfile, but got the same result as I did with `--cert`
```Dockerfile
ENV DENO_TLS_CA_STORE="system"
```

My suspicion is that the `--cert` arg as well as the `DENO_TLS_CA_STORE` env var both take effect for import resolution, but not for the download of the actual binary runtime when everything is bundled.

**Comment by w0otness** [comment]
Similar to this (but on windows), when doing `deno cache` with `import _ from 'npm:lodash'` (or any npm package) it doesn't matter whether -cert is specified or not, or the env var DENO_TLS_CA_STORE

> Sending fatal alert BadCertificate
> error: Error getting response at https://registry.npmjs.org/lodash
> 
> Caused by:
>     0: error sending request for url (https://registry.npmjs.org/lodash): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
>     1: error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
>     2: invalid peer certificate contents: invalid peer certificate: UnknownIssuer`

there is a chance it has commonality with #10312

**Comment by dsherret** [comment]
@w0otness that specific issue should be resolved in the latest version. This issue isn't though.

**Comment by dsherret** [comment]
Should be fixed in 1.29, but let me know if not.
